osbs/api.py

```diff
@@ -465,10 +474,11 @@ def _create_build_config_and_build(self, build_request):
             prev_version = existing_bc['status']['lastVersion']
             build_id = self.os.wait_for_new_build_config_instance(
                 build_config_name, prev_version)
-            build = BuildResponse(self.os.get_build(build_id).json())
+            build = BuildResponse(self.os.get_build(build_id).json(),
+                                  self.get_pod_for_build(build_id))
         else:
             response = self.os.start_build(build_config_name)
-            build = BuildResponse(response.json())
+            build = BuildResponse(response.json(), self.get_pod_for_build_json(response.json()))
```

Comment:
We've only just started the build here, so the pod might very well not yet exist. Won't that raise an exception in that case?

Reply:
Since its calling get_pod_for_build_json which is a wrapper for get_pod_for_build, it's easy enough to add a try: except: block and return None (which is valid) if the pod doesn't exist.

Reply:
You're right. Do we need a new API method for this?

Reply:
a new API method means that people calling BuildResponse.get_error_message have to pass that error and decide if they also need to call another function. Or if BuildResponse does it, then BuildResponse needs access to an OSBS instance to call the API. 

Based on twaugh's objections, I favor stashing the OSBS instance in the BuildResponse, which can then retrieve the PodResponse iff needed.

Reply:
It could even just be an optional parameter to the `get_error_message` method.

Reply:
I would prefer to not change get_error_message, because that means changing programs that call it. Changing the __init__ function is something that is mostly self-contained within osbs-client.

Reply:
fixed this so that BuildReponse takes an OSBS instance and only retrieves the PodResponse if needed. Code is much cleaner.